thread/stack-allocator.cpp

```diff
@@ -28,7 +28,10 @@ limitations under the License.
 
 namespace photon {
 
-const static size_t PAGE_SIZE = getpagesize();
+inline uint32_t get_pagesize() {
+    static uint32_t page_size = getpagesize();
+    return page_size;
```

Comment:
It's better not to use auto init, as it incurs an implicit mutex, which is useless and inflates the code size.

Simply init it manually as:
```cpp
static uint32_t page_size = 0;
if (unlikely(page_size == 0)) 
    page_size = getpagesize();
return page_size;
```

Reply:
Or rather, we can define an global variable that gets initialized in vcpu_init().